### ISSUE TITLE
Cancelling `next_msg()` does not cancel internal `_next_msg()` task

### ISSUE THREAD

**Issue by rnbguy** [nats-io/nats.py#296]
The following exception-handling doesn't handle `asyncio.CancelledError` which lets `_next_msg()` remain pending, even though the top level `next_msg()` is cancelled.

https://github.com/nats-io/nats.py/blob/4b6cdf2532e4851b5eece8721b1a782bcc322d0c/nats/aio/subscription.py#L157-L165

Probably it should be followed by,

```py
        except asyncio.CancelledError:
            future.cancel()
            task.cancel()
```

Ref: https://docs.python.org/3/library/asyncio-task.html#task-object

**Comment by rnbguy** [comment]
Code to reproduce this error,
```py
import asyncio
import nats

loop = asyncio.new_event_loop()

client = loop.run_until_complete(nats.connect("nats://demo.nats.io:4222"))

sub = loop.run_until_complete(client.subscribe("foo"))

task = loop.create_task(sub.next_msg(timeout=None))

try:
    fut = loop.run_until_complete(asyncio.wait_for(asyncio.shield(task), timeout=0.1))
except asyncio.TimeoutError:
    pass

task.cancel()  # this is supposed to cancel internal `_next_msg()` task, but it doesn't

loop.run_until_complete(client.close())
loop.close()
```

**Comment by wallyqs** [comment]
Fixed via #301
